src/H5Dcontig.c

```diff
@@ -615,13 +615,15 @@ H5D__contig_may_use_select_io(const H5D_io_info_t *io_info, H5D_io_op_type_t op_
     HDassert(dataset);
     HDassert(op_type == H5D_IO_OP_READ || op_type == H5D_IO_OP_WRITE);
 
+    dataset = io_info->dset;
```

Comment:
This should be immediately before the HDassert(dataset) line, otherwise that assert will always be tripped.

Reply:
Was building in release so missed this one! Just fixed the assertion and pushed.